crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -727,6 +727,13 @@ define_tedge_config! {
         path: Utf8PathBuf,
     },
 
+    db: {
+        /// The directory used to store any persistent data that survives tedge updates and even firmware updates
+        #[tedge_config(example = "/data/tedge", default(value = "/data/tedge"))]
+        #[doku(as = "PathBuf")]
+        path: Utf8PathBuf,
+    },
```

Comment:
Yeah I'll have to take some time to think about this one. We might need to have a discussion regarding the nuances between the two behaviours and what are the pros/cons.

Reply:
As discussed we have decided to put the entity store persistence under the `/etc/tedge/.tedge-mapper-<name>` folder (or something to that effect).

Having the data under `/etc/tedge` rather than the persisted `/data/tedge` folder makes the upgrade path of the agent across partitions more reliable, as the state under /etc/tedge will be left unmodified allow a rollback to the previous partition more possible (without changing the potentially upgraded state files).

Reply:
As discussed offline, reverted this config setting and persisting the log file to `/etc/tedge/.tedge-mapper-c8y` directory instead.